src/realm/parser/driver.cpp

```diff
@@ -347,6 +347,12 @@ std::unique_ptr<Subexpr> OperationNode::visit(ParserDriver* drv, DataType type)
             left = m_left->visit(drv);
         }
     }
+    if (!Mixed::is_numeric(left->get_type(), right->get_type())) {
```

Comment:
This solves the problem, but it also limits the feature to not work with mixed properties. Can we also support mixed and if there are any non-numeric values encountered, treat them like a null value? That would change the behaviour so that a query like the one in your test `age + spouse.name == 50` returns no matches instead of throwing. What do you think?

Reply:
Done. Quite easy to support that.